### PR TITLE
NO-ISSUE: Adapting tests to defined test strategy

### PR DESCRIPTION
**Pull request by mangelajo**
In https://github.com/flightctl/flightctl/pull/204 we defined an initial
test strategy and this test is pushing tests around to start adapting to
that model.

DB related tests are moved into integration, some task/reposync tests
are converted to unit-tests, including some small changes to enhance
testability of the code.

Signed-off-by: Miguel Angel Ajo Pelayo <majopela@redhat.com>

### CODE REVIEW

File: internal/tasks/resourcesync.go [hunk 1]
```diff
@@ -314,71 +314,16 @@ func (r *ResourceSync) updateResourceSyncStatus(rs *model.ResourceSync) {
...
-	var observedGen int64 = 0
-	if rs.Status.Data.ObservedGeneration != nil {
-		observedGen = *rs.Status.Data.ObservedGeneration
-	}
-	var prevHash string = util.DefaultIfNil(rs.Status.Data.ObservedCommit, "")
-	return hash != prevHash || observedGen != *rs.Generation
-}
-
-func isValidFile(filename string) bool {
+func IsValidFile(filename string) bool {
```
Comment (mangelajo) [inline internal/tasks/resourcesync.go:317]: undo this, this should just be unit testing

File: test/integration/tasks/resourcesync_test.go [hunk 2]
```diff
@@ -194,35 +197,36 @@ var _ = Describe("ResourceSync", Ordered, func() {
...
+			willRunSync = rs.NeedsSyncToHash("hash")
 			Expect(willRunSync).To(BeFalse())
 		})
+		// move to unit
 		It("File validation", func() {
-			Expect(isValidFile("something")).To(BeFalse())
-			Expect(isValidFile("something.pdf")).To(BeFalse())
-			for _, ext := range fileExtensions {
-				Expect(isValidFile(fmt.Sprintf("file.%s", ext))).To(BeTrue())
+			Expect(tasks.IsValidFile("something")).To(BeFalse())
```
Comment (mangelajo) [inline test/integration/tasks/resourcesync_test.go:226]: Just move to unit test

File: test/integration/tasks/resourcesync_test.go [hunk 3]
```diff
@@ -166,10 +167,12 @@ var _ = Describe("ResourceSync", Ordered, func() {
 				&ownedFleets[1],
 			}
 
-			delta := resourceSync.fleetsDelta(ownedFleets, newFleets)
+			delta := resourceSync.FleetsDelta(ownedFleets, newFleets)
```
Comment (mangelajo) [inline test/integration/tasks/resourcesync_test.go:170]: move to unit test.

File: Makefile [hunk 4]
```diff
@@ -31,6 +16,7 @@ help:
 	@echo "    tidy:            tidy go mod"
 	@echo "    lint:            run golangci-lint"
 	@echo "    build:           run all builds"
+	@echo "    integration-test: run integration tests"
```
Comment (avishayt) [inline Makefile:19]: Let's add a "test" target that runs all tests?

File: internal/store/model/resourcesync_test.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+			Generation: util.Int64ToPtr(1),
+		},
+		Spec: &JSONField[api.ResourceSyncSpec]{
+			Data: api.ResourceSyncSpec{
+				Repository: util.StrToPtr("demoRepo"),
+				Path:       util.StrToPtr("/examples"),
+			},
+		},
+	}
+	if obsHash != nil || obsGen != nil {
```
Comment (avishayt) [inline internal/store/model/resourcesync_test.go:75]: The code will panic if one of them is nil

**Review by avishayt — COMMENTED** [review]
Overall looks great!

File: internal/store/model/resourcesync_test.go (hunk 5 above)
Comment (mangelajo) [reply internal/store/model/resourcesync_test.go:75]: ack I will provide defaults when not provided.

**Comment by mangelajo** [comment]
Thanks for the review @avishayt , updating the patch.

File: Makefile (hunk 4 above)
Comment (mangelajo) [reply Makefile:19]: Done.

File: internal/store/model/resourcesync_test.go (hunk 5 above)
Comment (mangelajo) [reply internal/store/model/resourcesync_test.go:75]: Done.

**Review by avishayt — APPROVED** [review]

**Pull request merged**